data-pipeline/src/telemetry/mod.rs

```diff
@@ -87,13 +88,16 @@ impl TelemetryClientBuilder {
             self.language_version.unwrap(),
             self.tracer_version.unwrap(),
         );
+        builder.with_config(self.config);
+        // Send only metrics and logs and drop lifecycle events
+        builder.flavor = TelemetryWorkerFlavor::MetricsLogs;
 
         if let Some(id) = self.runtime_id {
             builder.runtime_id = Some(id);
         }
 
         let (worker, handle) = builder
-            .spawn_with_config(self.config)
+            .spawn()
```

Comment:
these two APIs `with_config` and `spawn_with_config`/`spawn` look every confusing, could add some docs which one to use and when?

Reply:
frankly we should just remove one, two configs all coming from manual code configuration is a bad API.

Although, I'm not going to block the PR for this.

Reply:
yeah I know, but I'm fixing the priority first, and then I think I'll add a builder pattern in a followup PR. The issue is that doing both at the same time makes the changes really hard to distinguish